gravitee-apim-gateway/gravitee-apim-gateway-core/src/test/java/io/gravitee/gateway/jupiter/core/v4/endpoint/loadbalancer/WeightedRandomLoadBalancerJCStressTest.java

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.gateway.jupiter.core.v4.endpoint.loadbalancer;
+
+import static org.mockito.Mockito.mock;
+
+import io.gravitee.definition.model.v4.endpointgroup.Endpoint;
+import io.gravitee.gateway.jupiter.core.v4.endpoint.ManagedEndpoint;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import org.openjdk.jcstress.Main;
+import org.openjdk.jcstress.annotations.Actor;
+import org.openjdk.jcstress.annotations.Expect;
+import org.openjdk.jcstress.annotations.JCStressTest;
+import org.openjdk.jcstress.annotations.Outcome;
+import org.openjdk.jcstress.annotations.State;
+import org.openjdk.jcstress.infra.results.IIII_Result;
+import org.openjdk.jcstress.infra.results.II_Result;
+
+// This is a concurrent stress tests that must be run independently
+@SuppressWarnings("java:S2187")
+public class WeightedRandomLoadBalancerJCStressTest {
```

Comment:
That's nice!
What about adding those in the CI? At least, could we add a comment to know how to start them manually?

Reply:
Actually the test result doesn't pass. The result needs human analysis to be sure the concurrent issue is not too high because it will exist anyway. (we don't want to introduce synchronized method here)